vignettes/plot_projection.Rmd

```diff
@@ -0,0 +1,283 @@
+Faceted plots can be created for multiple scenarios, locations, 
+forecast dates, models, etc. 
+Currently, the function can plot only quantiles data, with the possibility to 
+add "median" information from the model projections.
+
+For more information about the Hubverse standard format, please refer to the
+[HubDocs website](https://hubdocs.readthedocs.io/en/latest/format/tasks.html).
+
+The following vignette describes the principal usage of the 
+`plot_step_ahead_forecasts()` function.
+
+```{r lib}
+library(hubVis)
+```
+
+Plots are available in two output formats:
+
+- "interactive" format: a [Plotly](https://plotly.com) output object with interactive legend, hover
+text, zoom-in and zoom-out options, etc.
+- "static" format: a [ggplot2](https://ggplot2.tidyverse.org/) output object.
+By default, the output plot is "interactive", but it can be changed to "static"
+by setting  the `interactive` parameter to FALSE. See end of the document for examples.
+
+## Load and Filter Data
+
+The package contains 2 datasets that can be used for examples:
+
+- `example_round1.csv`: example of model output for a round associated with the 
+origin date: "2021-03-07" (called "round 1"), target: "incident case", for the
+US national level, from the
+[example-complex-scenario-hub](https://github.com/Infectious-Disease-Modeling-Hubs/example-complex-scenario-hub). The data set also contains an ensemble calculated by applying the 
+function: `hubEnsembles::simple_ensemble(df_round1, agg_fun = "median")`
+
+- `truth_data.csv`: example of target data from the [example-complex-scenario-hub](https://github.com/Infectious-Disease-Modeling-Hubs/example-complex-scenario-hub). The data here comes from the `"target-data/US_inc_case.csv"` file.
+
+### Load data
+
+```{r load_data}
+projection_path <- system.file("example_round1.csv", package = "hubVis")
+projection_data <- read.csv(projection_path)
+head(projection_data)
+
+truth_path <- system.file("truth_data.csv", package = "hubVis")
+truth_data <- read.csv(truth_path)
+head(truth_data)
+```
+
+### Data Preparation
+
+To fit the input format of the function, a `target_date` column needs to be 
+calculated for the model projections data.
+
+```{r data_prep}
+projection_data <- dplyr::mutate(
+  projection_data, target_date = as.Date(origin_date) + (horizon * 7) - 1)
+head(projection_data)
+```
+
+## Plot
+
+The function requires only 2 parameters:
+
+- `forecast_data`: a `model_output_df` object containing all the Hubverse
+ standard columns, including a `"target_date"`, and a `"model_id"` column in the 
+`hub_mdl_out_df format`. As all forecasts in forecast_data will be plotted, 
+ all filtering needs to happen outside this function.
+- ` truth_data`: a `data.frame` object containing the ground truth data, including 
+the columns: `time_idx` and `value`.
+
+### "Simple" plot
+
+The `"projection_data"` and `"truth_data"` contain information for multiple 
```

Comment:
Noting that there is inconsistent usage of quoting names of objects inside fixed-width font quotes. e.g.
`"projection_data"` vs `projection_data`.

I've tried to standardize to the convention of only the fixed-width font quotes for everything except for column names, where I include both. Although I'd be happy to follow a more concrete/written standard on this.

Reply:
Sorry for that! I will try to follow that standard in the future.